components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridConfigPersistenceManagerTest.java

```diff
@@ -0,0 +1,134 @@
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.database.utils.jdbc.NamedPreparedStatement;
+import org.wso2.carbon.identity.common.testng.WithCarbonHome;
+import org.wso2.carbon.identity.common.testng.WithH2Database;
+import org.wso2.carbon.identity.common.testng.WithRealmService;
+import org.wso2.carbon.identity.common.testng.WithRegistry;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.entitlement.EntitlementException;
+import org.wso2.carbon.identity.entitlement.cache.ConfigCache;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementConfigHolder;
+import org.wso2.carbon.identity.entitlement.internal.EntitlementServiceComponent;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import static org.wso2.carbon.identity.entitlement.PDPConstants.Algorithms.*;
+
+/**
+ * This class tests the behavior of the HybridConfigPersistenceManager class.
+ */
+@WithCarbonHome
+@WithRegistry(injectToSingletons = {EntitlementServiceComponent.class})
+@WithRealmService(injectToSingletons = {EntitlementConfigHolder.class}, initUserStoreManager = true)
+@WithH2Database(jndiName = "jdbc/WSO2IdentityDB", files = {"dbscripts/h2.sql"})
+public class HybridConfigPersistenceManagerTest {
+
+    private HybridConfigPersistenceManager hybridConfigPersistenceManager;
+    private RegistryConfigPersistenceManager registryConfigPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        hybridConfigPersistenceManager = new HybridConfigPersistenceManager();
+        registryConfigPersistenceManager = new RegistryConfigPersistenceManager();
+    }
+
+    @Test
+    public void testGetDefaultGlobalPolicyAlgorithmName() throws Exception {
+
+        String globalPolicyAlgorithmName = registryConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(globalPolicyAlgorithmName, DENY_OVERRIDES);
+    }
+
+    @DataProvider
+    public Object[][] globalPolicyAlgorithmData() {
+
+        return new Object[][]{
+                {DENY_OVERRIDES},
+                {PERMIT_OVERRIDES},
+                {FIRST_APPLICABLE},
+                {ONLY_ONE_APPLICABLE},
+                {ORDERED_DENY_OVERRIDES},
+                {ORDERED_PERMIT_OVERRIDES}
+        };
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testGetGlobalPolicyAlgorithmNameFromRegistry(String policyAlgorithm) throws Exception {
+
+        // Add the first global policy combining algorithm.
+        boolean addResult = registryConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+
+        String policyAlgorithmFromRegistry = hybridConfigPersistenceManager.getGlobalPolicyAlgorithmName();
```

Comment:
1. Shouldn't this line evaluated based on the addResult value ? 
2. What if it returns false ? 
3. Are there any test cases covering the. 'false' returning scenarios ?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/5905/commits/54e9c3520f918ba346019afa459cf6669a1b2360